### PR TITLE
refactor: Harmonize subnet variables

### LINKED ISSUE #419: Harmonize the subnet_id* variables
#411 raised the question of renaming the `subnet_ids_gitlab_runner` and `subnet_id_runners`. Very hard to understand what is configured here.

Proposal:
Introduce a new variable `subnet_id`. Use this subnet for the runner and the executors. I do not see any advantage of having the runner placed in another subnet as the executors. Both should be in the same subnet and the module should be deployed per availability zone to get the resilience needed.

Mark the other variables deprecated. But still use them if `subnet_id` is not filled.

```hcl
variable "subnet_id" {
  description = "Subnet used for the runner and the executors."
  type        = string
}

variable "subnet_id_runners" {
  description = "Deprecated. Use subnet_id instead. List of subnets used for hosting the gitlab-runners."
  type        = string
}

variable "subnet_ids_gitlab_runner" {
  description = "Deprecated. Use subnet_id instead. Subnet used for hosting the GitLab runner."
  type        = list(string)
}
```

### PR DESCRIPTION
**Pull request by kayman-mk**
## Description

Deprecates `subnet_ids_gitlab_runner` and `subnet_id_runners` and introduces `subnet_id` to make the module more readable as explained in #411 and #419. Thus it it no longer possible to place the runner and the executors in different subnets. For a resilient configuration deploy the module once per availability zone.

Closes #419

## Migrations required

No, but migration should be done.

Replace `subnet_ids_gitlab_runner` and `subnet_id_runners` by `subnet_id`. Please notice that it is no longer possible to place the runner and executors in different subnet. For a resilient configuration, I suggest to deploy the module once per availability zone.

## Verification

Checked the following:
- config.toml checked for subnet value using `subnet_id`. Success. Correct subnet used.
- Check that the Runner is created in correct subnet using `subnet_id`. Success.
- Remove `subnet_id` and use the old variables `subnet_ids_gitlab_runner` and `subnet_id_runners` again. `terraform plan` does not show any differences. Success.

## Documentation

We use [pre-commit](https://pre-commit.com/) to update the Terraform inputs and outputs in the documentation via [terraform-docs](https://github.com/terraform-docs/terraform-docs). Ensure you have installed those components.

### CODE REVIEW

**Comment by kayman-mk** [comment]
Missing:
- verification
- rewrite examples

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2017 Niek Palm
+Copyright (c) 2018 The terraform-docs Authors.
```
Comment (npalm) [inline LICENSE:3]: Can you revert this change, was it changed by terraform-docs?

File: variables.tf [hunk 2]
```diff
@@ -19,13 +19,19 @@ variable "vpc_id" {
   type        = string
 }
 
+variable "subnet_id" {
```
Comment (npalm) [inline variables.tf:22]: I agree to you that the current names of varialbes can be conflicting.

- `subnet_id_runners` is intended as the subnet in which the docker machine instancesa are created
- `subnet_ids_gitlab_runner` is the subnet to host the instance with gitlab agent.

As you mention in the PR this drops the option to deploy to different subnets.  I have no clue if some is doing that. But don't see really a case for that.

File: variables.tf [hunk 3]
```diff
@@ -19,13 +19,19 @@ variable "vpc_id" {
...
+  description = "Subnet used for the runner and executors."
+  type        = string
+  default     = ""
```
Comment (npalm) [inline variables.tf:25]: Setting the default value to `""` won't work with the conditon you use like `var.subnet_id ? .....` The conditon should be evaluate to a boolean expression.

**Review by npalm — CHANGES_REQUESTED** [review]
@kayman-mk thanks for making the module simpler and better. I have added some reviews comment. 

- First, I was not able to run an upgrade of the default example. The condition is not valid.
- As you see I have alwasy some doubt about dropping flexibility. Since it cn impact the current ussers. But I agree there is no use of having 2 subnets.

**Review by npalm — CHANGES_REQUESTED** [review]
@kayman-mk thanks for making the module simpler and better. I have added some reviews comment. 

- First, I was not able to run an upgrade of the default example. The condition is not valid.
- As you see I have alwasy some doubt about dropping flexibility. Since it cn impact the current ussers. But I agree there is no use of having 2 subnets.

**Comment by kayman-mk** [comment]
Just some simple changes to the code how it could work and to discuss it. I will fix everything else to get a running version.

File: LICENSE (hunk 1 above)
Comment (kayman-mk) [reply LICENSE:3]: Oh yes, of course. I didn't do that manually.

**Comment by kayman-mk** [comment]
Verification still open and to be done as soon as I am back to work. ;-)

**Comment by kayman-mk** [comment]
Verified

**Comment by npalm** [comment]
Great will check!

**Review by npalm — APPROVED** [review]
@kayman-mk thanks for making the module simpler, removing complexity one of the habits we should practice more!
